tableqa/column_types.py

```diff
@@ -122,17 +126,28 @@ def adapt(self, x):
         return self.values[idx]
 
 
-class Year(Integer):
-    def adapt(self, x, context=None, *args, **kwargs):
-        # TODO
+class Date(Integer):
+    def adapt(self, x, context=None, metric='year', *args, **kwargs):
         x = x.lower()
-        curr = datetime.datetime.now().year
-        if  "last year" in x or "previous year" in x:
-            return " {}".format(curr - 1)
-        elif "next year" in x:
-            return " {}".format(curr + 1)
-        elif "current year" in x or "this year" in x:
-            return "{}".format(curr)
+        limit = {
+            'year': [1000, 0],
+            'month': [13, 0],
+            'week': [53, 0],
+        }
+        if metric == 'year':
+            curr = datetime.datetime.now().year
+        elif metric == 'month':
+            curr = datetime.datetime.now().month
+        elif metric == 'week':
+            curr = datetime.datetime.now().isocalendar()[1]
+        else:
+            raise TypeError("Error invoking Date.adapt: Unexpected value for argument 'matrix' ")
+        if "last "+metric in x or "previous "+metric:
+            return "in {}".format(curr - 1)
+        elif "next "+metric in x or "next "+metric in x:
+            return "in {}".format(curr + 1)
+        elif "current "+metric in x or "this "+metric in x:
+            return "in {}".format(curr)
         else:
             allowed = [
```

Comment:
Make a list `months=["january","february" ....]` and  do the same for `weeks`. Concat the corresponding list to `allowed ` as per the conditional check for weeks or months.